src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private static final int NOT_FOUND_INDEX = -1;
+    private final K[] keyArr;
+    private final V[] valueArr;
+    private int size;
+
+    public StorageImpl() {
+        keyArr = (K[]) new Object[MAX_SIZE];
+        valueArr = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = indexOfKey(key);
+        if (index != NOT_FOUND_INDEX) {
+            valueArr[index] = value;
+            size--;
+        }
+        if (size < MAX_SIZE) {
+            keyArr[size] = key;
+            valueArr[size] = value;
+            size++;
+        } else {
+            throw new RuntimeException("Storage is full");
+        }
```

Comment:
once again we throw and do not use redundant `else`s
```suggestion
        if (size >= MAX_SIZE) {
            throw new RuntimeException("Storage is full");
        }
        keyArr[size] = key;
        valueArr[size] = value;
        size++;
```

Reply:
Thank you, understand it, wanted you ask about this on QnA, but you already explain it